http.go

```diff
@@ -27,6 +28,9 @@ type ErrorType struct {
 // WriteJSONResponse write to te response stream
 func WriteJSONResponse(w http.ResponseWriter, response *Response) {
 	w.Header().Set("Content-Type", "application/json")
+	for header, value := range response.Headers {
+		w.Header().Set(header, value)
```

Comment:
Saving the result of `w.Header()` before the first call would save lots of function calls